eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/activities/MicroFragmentHostActivity.java

```diff
@@ -87,9 +97,9 @@ protected void onSaveInstanceState(Bundle outState)
     {
         super.onSaveInstanceState(outState);
 
-        // TODO Possibly create mutable State abstraction that holds all state for the Activity.
-        // TODO Key should not be used like this:
-        outState.putParcelable(Keys.MICRO_FRAGMENT_HOST.name(), new ParcelableBox<>(mMicroFragmentHost));
+        new MutableBundle(outState)
```

Comment:
So that could be something like this:
```java
        outState.putParcelable("schedjoules.state",
                new BundleBuilder()
                        .with(Keys.MICRO_FRAGMENT_HOST, new ParcelableBox<>(mMicroFragmentHost))
                        .with(Keys.CATEGORIES, new CategoriesBox(new CategoriesCache(this)))
                        .build());
```

My problem with this is that we start to expose the details of our serialization mechanism, I would like to avoid calling `putParcelable` directly or probably calling anything on `Bundle`.
So there is that `"schedjoules.state"` string key here now, which is not a `Key`. We can make it one, but then we would call `.name()` on it, which is not how it is supposed to work.

This is how we read the value in `onCreate()` now:
```java
new Argument<>(Keys.CATEGORIES, savedInstanceState).get()
```
This is straightforward, in-line with how we use it for Activity Intent or Fragment args. No need to think. With the version above, I wouldn't be sure how to access the values.

So I think there is great value in not exposing how we put and retrieve values to and from Bundles, especially nesting, and just using `Key<T>` accordingly.

`MutableBundle` is not the best name, I agree. The other option I thought is putting back that `Parameter.into(Bundle)` interface..  not sure if that's better. That couldn't be chained.

How about `FluentBundle` for the class? We are basically wrapping a Bundle here for adapting to our serialization and also making it fluent, so it seems fitting.